python/ray/data/tests/test_dynamic_block_split.py

```diff
@@ -0,0 +1,63 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+import ray
+from ray.data.block import BlockMetadata
+from ray.data.datasource import Datasource
+from ray.data.datasource.datasource import ReadTask
+
+from ray.tests.conftest import *  # noqa
+
+
+def test_read_large_data(ray_start_cluster):
+    # Test 20G input with single task
+    num_batch = 20
+    ctx = ray.data.context.DatasetContext.get_current()
+    block_splitting_enabled = ctx.block_splitting_enabled
+    ctx.block_splitting_enabled = True
+
+    try:
+        cluster = ray_start_cluster
+        cluster.add_node(num_cpus=1)
+
+        ray.init(cluster.address)
+
+        # Data source generates multiple 1G random bytes data
+        class LargeBytesDatasource(Datasource):
+            def prepare_read(self, parallelism):
```

Comment:
This method is deprecated, prefer using create_reader().

Reply:
@jianoaix - yeah, forgot about it, updated.